timelock-impl/src/main/java/com/palantir/atlasdb/timelock/lock/watch/LockEventLogImpl.java

```diff
@@ -38,6 +43,13 @@
 import java.util.stream.Collectors;
 
 public class LockEventLogImpl implements LockEventLog {
+    private static final SafeLogger log = SafeLoggerFactory.get(LockEventLogImpl.class);
+
+    // Log at most 1 line every 2 minutes. Diagnostics are expected to be triggered
+    // on exceptional circumstances and a one-off basis. This de-duplicates when we
+    // need diagnostics on large clusters.
+    private static final RateLimitedLogger diagnosticLog = new RateLimitedLogger(log, 1 / 120.0);
+
```

Comment:
Do we specify this at API level?

Reply:
We could. Though, I would not personally document this further than the PR discussion. This endpoint should be rarely called